### DISCUSSION TITLE
Automatically downloading latest Eclipse integration build: How to get information if build is unstabel

### DISCUSSION THREAD

**Discussion by thahnen** [eclipse-platform/eclipse.platform.ui#1831] [Q&A] [answered]
Hey everyone 👋 

I need some help with the feasibility of the following matter: Currently, I'm building my own "version" of Eclipse once in a while for testing via the latest integration builds ([like this one](https://download.eclipse.org/eclipse/downloads/drops4/I20240418-1800/)). This is done by downloading the OS-specific version, unpacking it, and running the `org.eclipse.equinox.p2.director` application to install some features/plug-ins necessary for testing. It is just a primitive solution, but it works.

I'd like to automate this as the download link can be templated with the date and time from the integration build. The issue is that I missed the information on whether the integration build was unstable or not. The information is displayed on the [page of the specific build](https://download.eclipse.org/eclipse/downloads/drops4/I20240415-1800/) but I was hoping that I don't have to scrape this page in order to see if it was unstable or not.

So my question is: What did I miss, or is there actually another way to retrieve the information if the integration build is unstable or not (like from the build or somewhere else) as I try to automate it to only pick the latest, stable integration build?

Best,
Tobias

**Comment by fedejeanne** [comment]
Hello @thahnen,

Would this help (from https://download.eclipse.org/eclipse/downloads/)?

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/2205684/3338986a-81d2-428a-b9ca-a9dddb562795)

The information is here: 

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/2205684/86d648ad-6cca-41dd-b9a8-373e59512b8c)

**Reply by thahnen** [reply]
That was my idea of last resort, to scrape the HTML 👍 
I was hoping that maybe there would be a more convenient option ^^

**Reply by iloveeclipse** [reply]
Unstable builds aren't published to https://download.eclipse.org/eclipse/updates/4.32-I-builds/, so you can grab latest tag and check if the url https://download.eclipse.org/eclipse/updates/4.32-I-builds/<tag\> is there. Not sure if this is easier as parsing html.

**Reply by fedejeanne** [reply]
> Unstable builds aren't published to https://download.eclipse.org/eclipse/updates/4.32-I-builds/

But https://download.eclipse.org/eclipse/updates/4.32-I-builds/I20240415-1800/ was published and it was unstable... :-\ . I mean the URL exists.

**Reply by iloveeclipse** [reply]
@merks : I thought the unstable I-Builds update sites aren't created?

**Reply by akurtakov** [reply, marked as answer]
Update sites are always created (often one needs this update site to test breakage) but if unstable they are not added to the composite  https://download.eclipse.org/eclipse/updates/4.32-I-builds/ .
So best path is fetch https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.32-I-builds/compositeContent.jar parst the xml in it and use the latest I-build listed there - that's the last stable build.

**Reply by merks** [reply]
I have no clue how the update sites are managed.  I think they are always created, but it might be the case that they are not composed into the composite if they are not stable. But I can't be sure of that now because looking at this:

https://www.eclipse.org/downloads/download.php?file=/eclipse/updates/4.32-I-builds/compositeContent.jar

I only see these:

![image](https://github.com/eclipse-platform/eclipse.platform.ui/assets/208716/67d4c3d7-b1c2-4e9f-b9c9-b88dd1c6500c)

I think @MohananRahul would know the details better.

**Reply by merks** [reply]
Of course one can just use the the composite repo URL directly and that will install the latest...

**Reply by thahnen** [reply]
I will follow the approach with the `compositeContent.jar` and parsing the XML. Thank you everyone for the suggestions!

**Reply by merks** [reply]
It's none of my business, by why don't you just use the composite repository directly?  After all, the p2 director is just as happy to use a composite as it is to use a simple repository and the net effect is expected to be the same, i.e., the newest/latest versions will be installed...

**Reply by thahnen** [reply]
I will always start with Eclipse not installed - mostly for macOS - to see if the application files that are no direct Eclipse plug-ins / features are also working as well and are not randomly breaking on the newest integration builds.

Having an Eclipse installation already on my machine and only using the repository as the Eclipse Update Site it is won't provide me with these files if I'm not mistaken. I might be wrong, but that is my assumption ^^
I could also build everything from Maven/Tycho product but I haven't looked into that yet, as I said nowadays I'm only building it from time to time but I'm hoping to use the latest (stable) iBuild for development every day.
